src/NServiceBus.Core/Unicast/MessageHandlerRegistry.cs

```diff
@@ -23,20 +22,21 @@ internal MessageHandlerRegistry(Conventions conventions)
         }
 
         /// <summary>
-        ///     Gets the list of <see cref="IHandleMessages{T}" /> <see cref="Type" />s for the given
+        ///     Gets the list of handlers <see cref="Type" />s for the given
         ///     <paramref name="messageType" />.
         /// </summary>
-        public IEnumerable<Type> GetHandlerTypes(Type messageType)
+        public IEnumerable<MessageHandler> GetHandlersFor(Type messageType)
```

Comment:
Ok, I see now, you made the handler concept explicit, :+1: , disregard the previous comment about the method having the wrong name.